easybuild/framework/easyblock.py

```diff
@@ -2378,6 +2378,14 @@ def fetch_step(self, skip_checksums=False):
                 raise EasyBuildError("EasyBuild-version %s is newer than the currently running one. Aborting!",
                                      easybuild_version)
 
+        source_deps = self.cfg['source_deps']
+        pre_fetch_env = None
+        # load modules for source dependencies (if any)
+        if source_deps:
+            pre_fetch_env = copy.deepcopy(os.environ)
+            source_deps_mod_names = [d['short_mod_name'] for d in source_deps]
```

Comment:
@PetrKralCZ `short_mod_name` field is still `None` at this point, which causes the crash in `run_module`, so we're overlooking something when parsing dependencies I think...

Reply:
This problem is now fixed, via 0db8fd2